fetch-node.js

```diff
@@ -0,0 +1,24 @@
+'use strict';
+
+var wrapFetchForNode = function (fetch) {
+    // Support schemaless URIs on the server for parity with the browser.
+    // https://github.com/matthew-andrews/isomorphic-fetch/pull/10
+    return function (url, options) {
+        if (/^\/\//.test(url)) {
+            url = 'https:' + url;
+        }
+        return fetch.call(undefined, url, options);
+    };
+};
+
+module.exports = function (context) {
+    // Use .bind() as a side effect, which is that it clones. That way,
+    // even if we set `Promise`, we have no side effects on `node-fetch`.
+    var fetch = require('node-fetch').bind(null);
```

Comment:
Yes, that's right. This kind of implementation follows the same principle as a ponyfill, which is that loading a module should not have any side effects on global state.

If one bit of code `requires`s `fetch-ponyfill` and passes in a Promise implementation, and another bit of code doesn't, each bit of code will get exactly what it expects.

Ugh. Well, that was the intention. I just tested it, and it doesn't actually work. `node-fetch` [allows a custom promise](https://github.com/bitinn/node-fetch/blob/master/index.js#L39-L42), but it has to be set on the global `fetch` function. Will clean that up.